app/scripts/modules/core/src/slack/SlackReader.spec.ts

```diff
@@ -0,0 +1,33 @@
+import { ISlackChannel, SlackReader } from './SlackReader';
+
+const mockSlackChannels: ISlackChannel[] = [
+  {
+    id: '0-test',
+    name: 'testchannel',
+    is_channel: true,
+    creator: 'creator',
+    is_archived: false,
+    name_normalized: 'testchannel',
+    num_members: 25,
+  },
+  {
+    id: '1-test',
+    name: 'fakechannel',
+    is_channel: true,
+    creator: 'creator2',
+    is_archived: false,
+    name_normalized: 'fakechannel',
+    num_members: 25,
+  },
+];
+
+describe('SlackReader', () => {
+  it('should fetch a list of public Slack channels', () => {
+    spyOn(SlackReader, 'getChannels').and.returnValue(Promise.resolve(mockSlackChannels));
+
+    SlackReader.getChannels().then((channels: ISlackChannel[]) => {
```

Comment:
this test isn't really asserting anything useful.  I can replace `SlackReader` reference with an empty stub and the test still passes:

```js
it('should fetch a list of public Slack channels', () => {
    const foo = { getChannels() {} };
    spyOn(foo, 'getChannels').and.returnValue(Promise.resolve(mockSlackChannels));

    foo.getChannels().then((channels: any[]) => {
      expect(foo.getChannels).toHaveBeenCalled();
      expect(channels.length).toEqual(2);
    });
  });
```

The only useful thing to assert that I can think of is if `API.one` got called with `slack/channels`.

Reply:
Ok I will think about this more. I was trying to remove the dependency on the angular mocks. I assumed this would still call the API and on a successful response return the values, while on a failed response the test would fail.

Reply:
you should be able to do something like:
```
    const oneSpy = spyOn(API as any, 'one').and.callThrough();
    const getSpy = spyOn(API as any, 'getFn').and.returnValue(Promise.resolve(mockSlackChannels));
```
then you should be able to 
```
expect(oneSpy).toHaveBeenCalledWith('slack/channels);
```

... I think